src/components/app-content-switcher/AppContentSwitcher.tsx

```diff
@@ -1,12 +1,36 @@
+import { TypographyProps } from '@mui/material/Typography'
 import { Typography } from '@mui/material'
 import Tooltip from '@mui/material/Tooltip'
 import Switch from '@mui/material/Switch'
 import Stack from '@mui/material/Stack'
 import { defaultStyles } from '~/components/app-content-switcher/AppContentSwitcher.styles'
 
-const AppContentSwitcher = ({ active, handleChange, switchOptions, typographyVariant, styles }) => {
 
-  const renderBlock = (options, active) => (
+interface SwitcherProps {
+    active: boolean;
+    handleChange: () => void;
+    switchOptions: {
+        left?: {
+            text: string;
+            tooltip?: string;
+        };
+        right?: {
+            text: string;
+            tooltip?: string;
+        };
+    };
+    typographyVariant: TypographyProps['variant'];
+    styles?: object;
+}
+
+type SwitchOption = {
+    text: string,
+    tooltip?: string
+}
+
+const AppContentSwitcher = ({ active, handleChange, switchOptions, typographyVariant, styles }: SwitcherProps) => {
+    
```

Comment:
If you want to describe type for props, you need to use another syntax.
Like this: const AppContentSwitcher: FC<SwitcherProps> = ({ active, handleChange, switchOptions, typographyVariant, styles }) =>

But first of all import {FC} from 'react'

Reply:
refactored